analysis/image_info.py

```diff
@@ -0,0 +1,247 @@
+        
+        Arguments:
+            obs_code : string
+               The Observatory code.
+        """
+        self.obs_code = obs_code
+        self.obs_loc_set = True
+
+    def set_obs_position(self, lat, long, alt):
+        """
+        Manually set the observatory location and clear
+        the observatory code.
+        
+        Arguments:
+            lat : float - Observatory latitude.
+            long : float - Observatory longitude.
+            alt : float - Observatory altitude.
+        """
+        self.obs_code = ''
+        self.obs_lat = lat
+        self.obs_long = long
+        self.obs_alt = alt
+        self.obs_loc_set = True
+
+    def set_epoch(self, epoch):
+        """
+        Manually set the epoch for this image.
+        
+        Arguments:
+            epoch : astropy Time object.
+        """
+        self.epoch_ = epoch
+        self.epoch_set_ = True
+
+    def get_epoch(self):
+        """
+        Get the epoch for this image.
+        
+        Returns:
+            epoch : astropy Time object.
+        """
+        assert(self.epoch_set_)
+        return self.epoch_
+
+    def pixels_to_skycoords(self, pos):
+        """
+        Transform the pixel position within an image
+        to a SkyCoord.
+        
+        Arguments:
+            pos : pixel_pos
+                A pixel_pos object containing the x and y
+                coordinates on the pixel.
+
+        Returns:
+            A SkyCoord with the transformed location.
+        """
+        return self.wcs.pixel_to_world(pos.x, pos.y)
+
+    def approximate_radius(self):
+        """
+        Compute an approximate radius of the image.
+        
+        Arguments: NONE
+
+        Returns:
+            A radius in degrees.
+        """
+        corner = self.wcs.pixel_to_world(0.0, 0.0)
+        radius = self.center.separation(corner)
+        return radius
+
+    def ra_radius(self):
+        edge = self.wcs.pixel_to_world(0.0, self.height/2)
+        radius = self.center.separation(edge)
+        return radius
+
+    def dec_radius(self):
+        edge = self.wcs.pixel_to_world(self.width/2, 0.0)
+        radius = self.center.separation(edge)        
+        return radius
+
+
+class ImageInfoSet():
+    def __init__(self):
+        self.stats = []
+        self.num_images = 0
+        
+    def set_times_mjd(self, mjd):
+        """
+        Manually sets the image times.
+        
+        Arguments:
+            mjd : List of floats
+                Image times in MJD.
+        """
+        assert(len(mjd) == self.num_images)
+        for i in range(self.num_images):
+            self.stats[i].set_epoch(Time(mjd[i], format='mjd'))
+
+    def get_image_mjd(self, index):
+        """
+        Return the MJD of a single image.
+        
+        Argument:
+            index : integer
+                The index of the image.
+
+        Returns:
+            float : timestamp in MJD.
+        """
```

Comment:
Doesn't have a method body.

Reply:
I'm really not sure how that happened. Fixed.